tools/adventure-pack/src/iteratorPrototypeFilter.ts

```diff
@@ -0,0 +1,23 @@
+import { iteratorPrototype } from "./iteratorPrototype";
+import { iteratorToIterable } from "./iteratorToIterable";
+
+declare global {
+  interface Iterator<T> {
+    filter(
+      callbackFn: (element: T, index: number) => boolean,
```

Comment:
So requiring the filtering function to return a boolean is nice, but the regular Array filter doesn't actually require that, you can return anything and it gets interpreted as a boolean. Could we look up what TypeScript's built-in definitions for Array filter are, and use the same return type here?

Reply:
I ended up writing the params from this PR 

https://github.com/microsoft/TypeScript/pull/7779